src/index.d.ts

```diff
@@ -545,165 +1005,885 @@ declare namespace admin.messaging {
     launchImage?: string;
   }
 
+  /**
+   * Represents a critical sound configuration that can be included in the
+   * `aps` dictionary of an APNs payload.
+   */
   interface CriticalSound {
+
+    /**
+     * The critical alert flag. Set to `true` to enable the critical alert.
+     */
     critical?: boolean;
+
+    /**
+     * The name of a sound file in the app's main bundle or in the `Library/Sounds`
+     * folder of the app's container directory. Specify the string "default" to play
+     * the system sound.
+     */ 
     name: string;
+
+    /**
+     * The volume for the critical alert's sound. Must be a value between 0.0
+     * (silent) and 1.0 (full volume).
+     */ 
     volume?: number;
   }
 
+
+  /**
+   * A notification that can be included in {@link admin.messaging.Message}.
+   */
   interface Notification {
+    /**
+     * The title of the notification.
+     */
     title?: string;
+    /**
+     * The notification body
+     */
     body?: string;
   }
-
+  /**
+   * Represents the WebPush protocol options that can be included in an
+   * {@link admin.messaging.Message}.
+   */ 
   interface WebpushConfig {
+
+    /**
+     * A collection of WebPush headers. Header values must be strings.
+     *
+     * See [WebPush specification](https://tools.ietf.org/html/rfc8030#section-5)
+     * for supported headers.
+     */
     headers?: {[key: string]: string};
+
+    /**
+     * A collection of data fields.
+     */
     data?: {[key: string]: string};
+
+    /**
+     * A WebPush notification payload to be included in the message.
+     */
     notification?: WebpushNotification;
+
+    /**
+     * Options for features provided by the FCM SDK for Web.
+     */
     fcmOptions?: WebpushFcmOptions;
   }
 
+  /** Represents options for features provided by the FCM SDK for Web
+   * (which are not part of the Webpush standard).
+   */
   interface WebpushFcmOptions {
     link?: string;
   }
 
+  /**
+   * Represents the WebPush-specific notification options that can be included in
+   * {@link admin.messaging.WebpushConfig}. This supports most of the standard
+   * options as defined in the Web Notification
+   * [specification](https://developer.mozilla.org/en-US/docs/Web/API/notification/Notification).
+   */ 
   interface WebpushNotification {
+
+    /**
+     * Title text of the notification.
+     */
     title?: string;
+
+    /**
+     * An array of notification actions representing the actions
+     * available to the user when the notification is presented.
+     */
```

Comment:
Add descriptions for the child elements? Not sure if that's possible with anonymous types such as this.